### PR TITLE
LP-2712 Fix catalog page for user with not application

### PR DESCRIPTION
**Pull request by asadali145**

### CODE REVIEW

**Review by javeriaejaz-arbisoft — APPROVED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2031/workflows/workflows_id/jobs/2031/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
-------------
Total:   1 line
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/courseware_override/helpers.py [hunk 1]
```diff
@@ -33,7 +33,9 @@ def get_business_line_prereq_courses(user):
     """
     if not (
         user.is_authenticated and
-        user.application_hub.is_written_application_completed and
+        hasattr(user, 'application_hub') and
+        user.application_hub.is_prerequisite_courses_passed and
+        hasattr(user, 'application') and
         user.application.business_line
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/courseware_override/helpers.py:39]: Have we tried `user.is_authenticated and user.application_hub and user.application_hub. is_written_application_completed  .............`
I guess we can replace hasattr with some checks if that exists or not.
Also as this condition is quite complex now, can we please move it to a separate method with a meaningful name?

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/courseware_override/helpers.py (hunk 1 above)
Comment (asadali145) [reply openedx/adg/lms/courseware_override/helpers.py:39]: I have tried user.application and user.application_hub but that does not work. Let me add a method in applications model for this.

**Review by annas-usman — APPROVED** [review]

File: openedx/adg/lms/applications/helpers.py [hunk 2]
```diff
@@ -307,3 +307,22 @@ def get_courses_from_course_groups(course_groups, user):
             courses_list.append(multilingual_course.course)
 
     return courses_list
+
+
+def is_user_qualified_for_bu_prereq_courses(user):
+    """
+    Checks whether a user satisfies all the conditions to attempt business line prerequisite courses.
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/helpers.py:314]: Let's please describe those conditions here.

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2032/workflows/workflows_id/jobs/2032/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/courseware_override/helpers.py (100%)
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
-------------
Total:   15 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2033/workflows/workflows_id/jobs/2033/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/courseware_override/helpers.py (100%)
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
-------------
Total:   15 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Review by ahmed-faraz46 — APPROVED** [review]

**Pull request merged**